lib/utils.js

```diff
@@ -20,48 +21,58 @@ var dateTracker = {
     }
 
     return model;
-  },
-  parseDatesInModel: function(model) {
-    if (typeof model === 'object') {
-      if (typeof model._date_created !== 'undefined') {
+  }
+
+  static parseDatesInModel(model) {
+    if (typeof model === `object`) {
+      if (typeof model._date_created !== `undefined`) {
         model.date_created = model._date_created;
         delete model._date_created;
       }
-      if (typeof model._date_updated !== 'undefined') {
+      if (typeof model._date_updated !== `undefined`) {
         model.date_updated = model._date_updated;
         delete model._date_updated;
       }
     }
 
     return model;
-  },
-  // `isModel` indicates whether the data passed in is a Bookshelf model or not
-  convertToISOStrings: function(isModel) {
-    return function(data) {
-      var created = isModel ? data.get('date_created') : data.date_created;
-      var updated = isModel ? data.get('date_updated') : data.date_updated;
-
-      if (isDate(created)) {
-        if (isModel) {
-          data.set('date_created', created.toISOString());
-        } else {
-          data.date_created = created.toISOString();
-        }
-      }
+  }
 
-      if (isDate(updated)) {
-        if (isModel) {
-          data.set('date_updated', updated.toISOString());
-        } else {
-          data.date_updated = updated.toISOString();
-        }
-      }
+  static extractDate(data, property, isModel) {
+    return isModel ? data.get(property) : data[property];
+  }
+
+  static convertToISOStrings(data) {
+    const created = DateTracker.extractDate(data, `date_created`);
```

Comment:
what inputs are you provisioning for here? It looks like the function starts uncertain what it's working on, but then later in the function sets values as if `data` is always a generic object